src/OmniSharp.MSBuild/ProjectSystem.cs

```diff
@@ -137,7 +137,25 @@ private IEnumerable<string> GetInitialProjectPaths()
             return _fileSystemHelper.GetFiles("**/*.csproj");
         }
 
-        private IEnumerable<string> GetProjectPathsFromSolution(string solutionFilePath)
+        // enumerating project file paths in MSBuild.SolutionSdk project.
+        private IEnumerable<string> GetProjectPathsFromSolutionBySolutionSdk(string solutionFilePath)
+        {
+            IEnumerable<string> CollectFunc(string slnproj)
+            {
+                // to avoid enumrating unsupported project.
+                foreach (var path in SolutionSdkFileUtil.GetEvaluatedProjectFilePaths(slnproj, _propertyOverrides, new string[] { ".csproj", ".fsproj", ".vbproj" }))
```

Comment:
OmniSharp only supports '.csproj'. '.vbproj' and '.fsproj' shouldn't be included.